src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepDynamicContext.java

```diff
@@ -90,12 +91,33 @@ void resume(StepContext context) throws Exception {
         if (executor != null) {
             throw new IllegalStateException("Already resumed");
         }
-        Queue.Item item = Queue.getInstance().schedule2(task, 0).getItem();
-        if (item == null) {
-            // TODO should also report when !ScheduleResult.created, since that is arguably an error
-            throw new IllegalStateException("queue refused " + task);
+        // If Jenkins restarts or crashes while we are waiting below, on the next startup the task may already be in
+        // the queue, in which case we should reuse it instead of scheduling a second task.
```

Comment:
Hmm, `Queue` scheduling should already reuse items (`ScheduleResult.Existing`) in most cases. You are now implementing `equals` and `hashCode` (see `duplicatesInQueue` logic with `liveGetItems` ) so is this logic still needed?

Reply:
Indeed it can be simplified, see [456010d](https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/344/commits/456010d9c970e0e2ee516d5f07b63dd5425f45ca).